internal/uploader/events_uploader_test.go

```diff
@@ -44,6 +45,169 @@ const (
+			events, err := ExtractEvents(nil)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(events)).To(Equal(0))
+		})
+	})
+
+	When("the reader is empty", func() {
+		BeforeEach(func() {
+			buffer = &bytes.Buffer{}
+		})
+
+		It("should not fail", func() {
+			events, err := ExtractEvents(buffer)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(events)).To(Equal(0))
+		})
+	})
+
+	When("the reader contains random data", func() {
+		BeforeEach(func() {
+			buffer = bytes.NewBufferString("whatever")
+		})
+
+		It("should fail", func() {
+			_, err := ExtractEvents(buffer)
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	var (
+		gz *gzip.Writer
+		tw *tar.Writer
+	)
+
+	When("the reader contains valid data", func() {
+		BeforeEach(func() {
+			buffer = &bytes.Buffer{}
+			gz = gzip.NewWriter(buffer)
+			tw = tar.NewWriter(gz)
+		})
+
+		Context("of 1 cluster", func() {
+			clusterID := strfmt.UUID("cluster1")
+
+			var config Config
+
+			When("the config is incomplete", func() {
+				BeforeEach(func() {
+					config = Config{
+						DeploymentType: "depType",
+						Versions: versions.Versions{
+							SelfVersion:    "1.2.3",
+							InstallerImage: "4.5.6",
+						},
+					}
+
+					metadataFile(tw, &clusterID, config)
+
+					Expect(tw.Close()).To(Succeed())
+					Expect(gz.Close()).To(Succeed())
+				})
+
+				It("should find this single metadata information for this cluster", func() {
+					events, err := ExtractEvents(buffer)
+
+					Expect(err).NotTo(HaveOccurred())
+					Expect(len(events)).To(Equal(1))
+					Expect(events[0]).To(Equal(generateExpectedEvents(clusterID.String(), config)))
+				})
+			})
+
+			When("the config is complete", func() {
+				BeforeEach(func() {
+					config = Config{
+						DeploymentType:         "depType",
+						DeploymentVersion:      "depVersion",
+						AssistedServiceVersion: "abcdef",
+						Versions: versions.Versions{
+							SelfVersion:     "1.2.3",
+							AgentDockerImg:  "4.5.6",
+							InstallerImage:  "7.8.9",
+							ControllerImage: "a.b.c",
+						},
+					}
+
+					metadataFile(tw, &clusterID, config)
+
+					Expect(tw.Close()).To(Succeed())
+					Expect(gz.Close()).To(Succeed())
+				})
+
+				It("should find this single metadata information for this cluster", func() {
+					events, err := ExtractEvents(buffer)
+
+					Expect(err).NotTo(HaveOccurred())
+					Expect(len(events)).To(Equal(1))
+					Expect(events[0]).To(Equal(generateExpectedEvents(clusterID.String(), config)))
+				})
+			})
+		})
+
+		Context("of 2 different clusters", func() {
+			clusterID1 := strfmt.UUID("cluster1")
+			clusterID2 := strfmt.UUID("cluster2")
+
+			var config1, config2 Config
+
+			When("the configs are complete", func() {
+				BeforeEach(func() {
+					config1 = Config{
+						DeploymentType:         "depType1",
+						DeploymentVersion:      "depVersion1",
+						AssistedServiceVersion: "abcdef1",
+						Versions: versions.Versions{
+							SelfVersion:     "1.2.31",
```

Comment:
nit: IMO it would be good to use more realistic data to give an idea to the reader what kind of data to expect.